File: practice15/12.cpp
//https://www.nowcoder.com/practice/9240357eefcf4d938b90bdd5eec3712b?tpId=90&tqId=30795&tPage=1&rp=1&ru=/ta/2018test&qru=/ta/2018test/question-ranking
#include<iostream>
#include<string>
#include<algorithm>
using namespace std;
int main()
{
	string str;
	int i, j, m, n, count, k, sum;
	char ch;
	while (cin >> str >> k)
	{
		int a[26] = { 0 };
		m = str.size();
		for (i = 0; i < m; i++)
		{
			if (str[i] == '0')
			{
				continue;
			}
			ch = str[i];
			count = 0;
			for (j = i; j < m; j++)
			{
				if (str[j] == '0')
				{
					continue;
				}
				if (str[j] == ch)
				{
					count++;
					str[j] = '0';
				}
			}
			n = ch - 'a';
			a[n] = count;
		}
		sort(a, a + 26);
		while (k > 0)    //不断将最大数减去一
		{
			a[25] = a[25] - 1;
			k = k - 1;
			sort(a, a + 26);
		}
		sum = 0;
		for (i = 0; i < 26; i++)
		{
			if (a[i] != 0)
			{
				sum = sum + a[i] * a[i];
			}
		}
		cout << sum << endl;
	}
	system("pause");
	return 0;
}

File: practice33/12.java

//https://www.nowcoder.com/practice/e27b9fc9c0ec4a17a5064fb6f5ab13e4?tpId=85&tqId=29883&tPage=3&rp=3&ru=/ta/2017test&qru=/ta/2017test/question-ranking
import java.math.BigInteger;
import java.util.*;
import java.util.Scanner;

public class Main {
    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in);
        int x=sc.nextInt();
        int y=sc.nextInt();
        int sum=x+y;
        BigInteger m=new BigInteger("1");
        BigInteger n=new BigInteger("1");
        for(Integer i=sum;i>sum-x;i--){
            m=m.multiply(new BigInteger(i.toString()));
        }
        for(Integer i=1;i<=x;i++){
            n=n.multiply(new BigInteger(i.toString()));
        }
        System.out.println(m.divide(n));
    }
}

